Use Case: Map Timeline Event View (#204)
ElektraInitiative/PermaplanT: doc/usecases/map_timeline_event_view.md

# Use Case: Map Timeline Event View

## Summary

- **Scope:** Map View
- **Level:** User Goal
- **Actors:** App User
- **Brief:** The user can see the addition/removal of elements as events using the timeline feature.
- **Status:** Draft

## Scenarios

- **Precondition:**
  The user has opened the app and selected a map.
- **Main success scenario:**
  The user sees events represented on the scrollbar of the timeline feature.
  Events include:
    - addition of an element to the map
    - deletion of an element on the map

> - **Alternative scenario:**

Author: @NurramoX 
> The user filters the timeline to only show e.g. events related to the addition of elements to the map.

This is great 👍 

> The user selects a date range on the timeline for changes made during a specific period.

I think the event view and date range selection is on the same UI element. This use case is covered in another PR

> If no events occurred, the app displays a message indicating the absence of events.

I prefer a less intrusive indicator. What do you think about adding a red outline around the timeline?

> - **Alternative scenario:**

Reviewer: > I think the event view and date range selection is on the same UI element. This use case is covered in another PR

Ah yes, I saw that after I made the review sorry.

> I prefer a less intrusive indicator. What do you think about adding a red outline around the timeline?

I would rather we use a red border as some kind of alert. I think that a lot of people of would perceive an issue when they see something red even though all is well. 
Instead we can present a lighthearted message right above the timeline such as, "It seems rather quiet in here."

> - **Alternative scenario:**

Author: @NurramoX don't worry :)
> I would rather we use a red border as some kind of alert. I think that a lot of people would perceive an issue when they see something red even though all is well.
Instead we can present a lighthearted message right above the timeline such as, "It seems rather quiet in here."

Uh, I was thinking about a dialog message, and yes, you are right^^  
Another possibility would be to integrate this message into the timeline view, like a toast message but overlayed on the timeline view.

- **Error scenario:**
  There is an error in the timeline display or event loading functionality. 
  In this case, the app displays an error message and allows the user to try again.
- **Postcondition:**
  The user successfully sees the events on the timeline scrollbar.

> - **Non-functional Constraints:**

Author: I added it for this use case as well, because it was present on the initial timeline use case.
The other use case requires about the same data then this use case.
I will also take a guess on how it will work. Upon adding/deleting an element, the changes are saved to both the local and the remote databases. If the connection is interrupted the changes can still be tracked by the local database and then synchronized upon reconnecting.
If a fresh user joins the map, all the event and timeline data is synchronized to her local database. 
How performant that is will be a part of my thesis ^^

> - **Non-functional Constraints:**

Reviewer: I understand, thanks :+1:

# Use Case: Map Timeline Event View

## Summary

- **Scope:** Map View
- **Level:** User Goal
- **Actors:** App User

> - **Brief:** The user can see the addition/removal of elements as events using the timeline feature, allowing them to understand how the map changed over time.

Reviewer 2: Brief should be brief:

```suggestion
- **Brief:** The user sees addition/removal of elements as events using the timeline feature.
```

- **Status:** Draft

## Scenarios

- **Precondition:**
  The user has opened the app and selected a map.
- **Main success scenario:**
  - The user sees events represented on the scrollbar of the timeline feature.
  Events include:
    - addition of an element to the map
    - deletion of an element from the map

>   - The timeline can be scaled with the mouse wheel, adjusting it's start and end date, thus showing more events of the future and the past.

Reviewer 2: ```suggestion
  - The timeline can be scaled, adjusting its start and end date, thus showing more events of the future and the past.
```
